server/src/server.rs

```diff
@@ -579,6 +600,36 @@ impl Server {
         server.nodes.check_timeouts(SESSION_TIMEOUT);
     }
 
+    async fn forward_resumer(&self) {
+        let mut interval = time::interval(Duration::new(SESSION_CLEANER_INTERVAL, 0));
+        loop {
+            interval.tick().await;
+            let s = self.clone();
+            s.check_resume_forwarding().await;
+        }
+    }
+
+    async fn check_resume_forwarding(&self) {
+        let mut server = self.state.write().await;
+        let clock = DefaultClock::default();
+        let rs_fwd = server.resume_forwarding.clone();
```

Comment:
Clone `self.state` instead, acquire the lock and drop it immediately when not needed (e.g. use a nested block or `drop(...)`